### ISSUE TITLE
Simple single layer line fit does not converge for most sets of sample points

### ISSUE THREAD

**Issue by TheFibonacciEffect** [FluxML/Flux.jl#2623]
I am following the [simple example of fitting a line](https://fluxml.ai/Flux.jl/stable/guide/models/overview/).

I used slightly different sample points than in the tutorial and the training completly diverges to infinity. Here is the working code that is exactly as in the tutorial:

```jl
using Flux
using Statistics

Flux.Random.seed!(42)
f(x) = 4x + 2

m = Dense(1=>1)
loss(model, x, y) = mean(abs2.(model(x) .- y));

using Flux: train!
opt = Descent()
x = hcat(0:5 ...)
d = [(x, f.(x))]
train!(loss, m, d,opt); loss(m, x, f.(x))
m.weight, m.bias
for epoch in 1:200
    train!(loss, m, d, opt)
end
@show loss(m, x, f.(x))
```

however when I use a slightly different set of sample points, for example `x = hcat(1:6 ...)` or `x = hcat(0:6 ...)` or `x = hcat(1:10 ...)` or anything else than in the tutorial the loss from the `train!` function becomes larger and larger diverging to infinity.

I would expect the training to be more stable, especially considering that this is an extremly simple model.
I know that what I would need to do is to change the learning rate, I guess. But maybe the default learning rate is set too high? Because this is really the intruductory tutorial.

(`-1:5` converges, `-1:6` diverges, `0:8` diverges, ...)

**Comment by mcabbott** [comment]
It's more robust with `Descent(0.01)`, maybe that would be better?

What's happening BTW is that each update pushes it to the other side of the valley, and further up the hill, resulting in an even bigger step next time. Saving the weight during training to see this:
```julia
julia> opt = Descent()  # as before
Descent(0.1f0)

julia> x = hcat(0:8 ...)  # suggested bad case
1×9 Matrix{Int64}:
 0  1  2  3  4  5  6  7  8

julia> store = [m.weight[1]]
1-element Vector{Float32}:
 16.1275

julia> for epoch in 1:200
           train!(loss, m, d, opt)
           push!(store, m.weight[1])
       end
ERROR: DomainError with Loss is Inf on data item 1, stopping training:
...

julia> store
32-element Vector{Float32}:
      16.1275
     -39.47737
     164.88016
    -586.4607
    2175.6436
   -7978.757
   29351.898
 -107887.15
  396645.84
...
```

**Comment by TheFibonacciEffect** [comment]
Ah okay, I see. Would it make sense to reduce the default value in the Descent optimizer to something smaller? You can always increase it if you want training to go faster, but having a lower default makes sure that it will converge.
